linux/LinuxProcess.c

```diff
@@ -707,15 +707,18 @@ static void LinuxProcess_writeField(const Process* this, RichString* str, Proces
       return;
    }
    case PROC_COMM: {
+      const char* procComm;
       if (lp->procComm) {
          attr = CRT_colors[Process_isUserlandThread(this) ? PROCESS_THREAD_COMM : PROCESS_COMM];
-         /* 15 being (TASK_COMM_LEN - 1) */
-         xSnprintf(buffer, n, "%-15.15s ", lp->procComm);
+         procComm = lp->procComm;
       } else {
          attr = CRT_colors[PROCESS_SHADOW];
-         xSnprintf(buffer, n, "%-15.15s ", Process_isKernelThread(lp) ? kthreadID : "N/A");
+         procComm = Process_isKernelThread(lp) ? kthreadID : "N/A";
       }
-      break;
+      /* 15 being (TASK_COMM_LEN - 1) */
+      int c = RichString_appendnWide(str, attr, procComm, MINIMUM(15, strlen(procComm)));
+      RichString_appendChr(str, ' ', 16 - c);
+      return;
    }
    case PROC_EXE: {
```

Comment:
`PROC_EXE` is affected from this issue just the same. as is the `CWD` column.

Similar may also apply for the DragonFlyBSD/FreeBSD `JAIL` column, and Solaris `ZONE` column.